src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -520,6 +540,14 @@ protected override AuthenticationTicket AuthenticateCore()
             }
         }
 
+        protected virtual IdentityModel.Clients.ActiveDirectory.AuthenticationResult RedeemAuthorizationCode(string authorizationCode)
```

Comment:
Then rename this middleware to `ActiveDirectoryAuthenticationMiddleware` and let the community offer a real provider-agnostic OpenID Connect middleware :smile: 

Out of curiosity, what's the advantage of using ADAL here?